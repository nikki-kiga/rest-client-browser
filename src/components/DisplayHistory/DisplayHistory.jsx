import React from 'react';
import PropTypes from 'prop-types';
import styles from './DisplayHistory.css';

const DisplayHistory = ({ history }) => {
  console.log(history);
  console.log('first', history[0]);
  const historyList = history.map(search => (
    <li key={search.url}className={search.method}>
      <span className={styles.method}>{search.method}</span>
      <button className={styles.url}>{search.url}</button>
    </li>
  ));

  return (
    <ul className={styles.history}>
      <p>This is a test</p>
      {historyList}
    </ul>
  );

};

DisplayHistory.propTypes = {
  history: PropTypes.arrayOf(PropTypes.shape({
    method: PropTypes.string,
    url: PropTypes.string
  }))
};

export default DisplayHistory;
