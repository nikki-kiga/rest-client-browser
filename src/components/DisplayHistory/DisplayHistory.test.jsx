import React from 'react';
import { shallow } from 'enzyme';
import DisplayHistory from './DisplayHistory.jsx';

describe('test display history component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<DisplayHistory history={[{ method:'get', url: 'www.test.com/api/test' }, { method:'post', url: 'www.test.com/api/test' }]}  />);
  });

  it('renders a display history component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
