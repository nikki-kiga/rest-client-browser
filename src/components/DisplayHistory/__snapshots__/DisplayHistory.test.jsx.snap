// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test display history component renders a display history component 1`] = `
<ul
  className="history"
>
  <p>
    This is a test
  </p>
  <li
    className="get"
    key="www.test.com/api/test"
  >
    <span
      className="method"
    >
      get
    </span>
    <button
      className="url"
    >
      www.test.com/api/test
    </button>
  </li>
  <li
    className="post"
    key="www.test.com/api/test"
  >
    <span
      className="method"
    >
      post
    </span>
    <button
      className="url"
    >
      www.test.com/api/test
    </button>
  </li>
</ul>
`;
